cmd/conduit/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/indexer/conduit"
+	"github.com/algorand/indexer/config"
+)
+
+import (
+	_ "github.com/algorand/indexer/exporters/all"
+	_ "github.com/algorand/indexer/importers/all"
+	_ "github.com/algorand/indexer/processors/all"
+)
+
+var (
+	logger          *log.Logger
+	defaultLogLevel string
+)
+
+// init() function for main package
+func init() {
+
+	// Setup logger
+	logger = log.New()
+
+	formatter := conduit.PluginLogFormatter{
+		Formatter: &log.JSONFormatter{
+			DisableHTMLEscape: true,
+		},
+		Type: "Conduit",
+		Name: "main",
+	}
+
+	logger.SetFormatter(&formatter)
+	logger.SetOutput(os.Stdout)
+	defaultLogLevel = "info"
+}
+
+// conduitCmd creates the main cobra command, initializes flags, and viper aliases
+func conduitCmd() *cobra.Command {
+	cfg := &conduit.Config{}
+	conduitCmd := &cobra.Command{
+		Use:   "conduit",
+		Short: "run the conduit framework",
+		Long:  "run the conduit framework",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// Tie cobra variables into local go-lang variables
```

Comment:
Are we sure we want to use environment variables? How does that work if there is more than one level in the YAML file?

Reply:
I think we need to modify it anyway since the function uses the `INDEXER` prefix which isn't what we want.

Reply:
For this level, we would only be setting the data directory via environmental variables.  All other (nested) configurations would be handled by the supplied config file `conduit.yml` and wouldn't be affected by environmental variables.